apps/core/src/utils/transaction/helpers/getBalanceChangeSummary.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+    type DryRunTransactionBlockResponse,
+    type SuiAddress,
+    type SuiTransactionBlockResponse,
+} from '@mysten/sui.js';
+
+export type BalanceChangeSummary = {
+    coinType: string;
+    amount: string;
+    recipient?: SuiAddress;
+    owner?: SuiAddress;
+};
+
+export const getBalanceChangeSummary = (
+    transaction: DryRunTransactionBlockResponse | SuiTransactionBlockResponse
+) => {
+    const { balanceChanges, effects } = transaction;
+    if (!balanceChanges || !effects) return null;
+
+    const negative = balanceChanges
+        .filter(
+            (balanceChange) =>
+                typeof balanceChange.owner === 'object' &&
+                'AddressOwner' in balanceChange.owner &&
+                BigInt(balanceChange.amount) < 0n
+        )
+        .map((balanceChange) => {
+            const amount = BigInt(balanceChange.amount);
+
+            // get address owner
+            const owner =
+                (typeof balanceChange.owner === 'object' &&
+                    'AddressOwner' in balanceChange.owner &&
+                    balanceChange.owner.AddressOwner) ||
+                undefined;
+
+            // find equivalent positive balance change
+            const recipient = balanceChanges.find(
+                (bc) =>
+                    balanceChange.coinType === bc.coinType &&
+                    BigInt(balanceChange.amount) === BigInt(bc.amount) * -1n
+            );
+
+            // find recipient address
+            const recipientAddress =
+                (recipient &&
+                    typeof recipient.owner === 'object' &&
+                    'AddressOwner' in recipient.owner &&
+                    recipient.owner.AddressOwner &&
+                    recipient.owner.AddressOwner) ||
+                undefined;
+
+            return {
+                coinType: balanceChange.coinType,
+                amount: amount.toString(),
+                recipient: recipientAddress,
+                owner,
+            };
+        });
+
+    const positive = balanceChanges
+        .filter(
+            (balanceChange) =>
+                typeof balanceChange.owner === 'object' &&
```

Comment:
Seems like these two conditions are consistent, so you could pre-filter the list so that it's more obvious that this is _also_ a positive / negative filter